packages/discovery-react-components/src/components/StructuredQuery/utils/getProjectFields.ts

```diff
@@ -0,0 +1,14 @@
+import DiscoveryV2 from 'ibm-watson/discovery/v2';
+
+export const getProjectFields = (
+  response: DiscoveryV2.ListFieldsResponse | null
+): string[] | null => {
+  if (response && response.fields) {
+    let fieldObjects = response.fields;
+    fieldObjects = fieldObjects.filter(fieldObject => fieldObject.field);
```

Comment:
nit - couldn't this be `reduce` instead of looping twice?
```suggestion
    const fieldNames = response.fields.reduce((accumulator, fieldObject) => {
      if (fieldObject.field) {
        accumulator.push(fieldObject.field);
      }
      return accumulator;
    }, []);
```